backend/src/api/model/series.rs

```diff
@@ -11,71 +11,124 @@ use crate::{
-    state: State,
-    title: Option<String>,
-    description: Option<String>,
 }
 
-impl Node for Series {
+pub(crate) struct WaitingSeries {
+    pub(crate) shared: SharedData,
+}
+
+impl Series for WaitingSeries {
     fn id(&self) -> Id {
-        Id::series(self.key)
+        Id::series(self.shared.key)
+    }
+
+    fn opencast_id(&self) -> &str {
+        &self.shared.opencast_id
     }
 }
 
-#[graphql_object(Context = Context, impl = NodeValue)]
-impl Series {
+#[graphql_object(Context = Context, impl = SeriesValue)]
+impl WaitingSeries {
     fn id(&self) -> Id {
-        Node::id(self)
+        Series::id(self)
     }
 
-    fn state(&self) -> State {
-        self.state
+    fn opencast_id(&self) -> &str {
+        self.opencast_id()
     }
 
-    fn title(&self) -> Option<&str> {
-        self.title.as_deref()
+    async fn host_realms(&self, context: &Context) -> ApiResult<Vec<Realm>> {
+        Series::host_realms(self, context).await
     }
+}
 
-    fn description(&self) -> Option<&str> {
-        self.description.as_deref()
+pub(crate) struct ReadySeries {
+    pub(crate) shared: SharedData,
+    title: String,
+    description: Option<String>,
+}
+
+impl Series for ReadySeries {
+    fn id(&self) -> Id {
+        Id::series(self.shared.key)
+    }
+
+    fn opencast_id(&self) -> &str {
+        &self.shared.opencast_id
+    }
+}
+
+#[graphql_object(Context = Context, impl = SeriesValue)]
+impl ReadySeries {
+    fn id(&self) -> Id {
+        Series::id(self)
     }
 
     fn opencast_id(&self) -> &str {
-        &self.opencast_id
+        self.opencast_id()
+    }
+
+    fn title(&self) -> &str {
+        &self.title
+    }
+
+    fn description(&self) -> Option<&str> {
+        self.description.as_deref()
     }
 
     #[graphql(arguments(order(default = Default::default())))]
     async fn events(&self, order: EventSortOrder, context: &Context) -> ApiResult<Vec<Event>> {
-        Event::load_for_series(self.key, order, context).await
+        Event::load_for_series(self.shared.key, order, context).await
     }
 
-    /// Returns a list of realms where this series is referenced (via some kind of block).
     async fn host_realms(&self, context: &Context) -> ApiResult<Vec<Realm>> {
-        let cols = Realm::col_names("realms");
-        let query = format!("\
-            select {cols} \
-            from realms \
-            inner join blocks \
-                on realms.id = blocks.realm_id and
-                type = 'series' and blocks.series_id = $1 \
-        ");
-        context.db.query_mapped(&query, dbargs![&self.key], Realm::from_row)
-            .await?
-            .pipe(Ok)
+        Series::host_realms(self, context).await
     }
 }
 
-impl Series {
+impl<S: Series> Node for S {
+    fn id(&self) -> Id {
+        Series::id(self)
+    }
+}
+
+impl SeriesValue {
+    // TODO Should this really return all of them?
     pub(crate) async fn load_all(context: &Context) -> ApiResult<Vec<Self>> {
```

Comment:
My idea was to filter the waiting series in the backend already, but we already do it in the frontend now so :shrug: 

And yes, it's only used in the editor. It will probably go OR if we find another use for it, it will probably need to evolve anyway, so yes, let's keep it as is and throw out the `TODO`.

Reply:
Ah ye ok.